src/pages/auth/components/SignUp.jsx

```diff
@@ -1,15 +1,42 @@
 import { useState } from "react";
+import { useAuth } from "../../../context";
 
 export function SignUp({ setAuthVal }) {
   const [signUpDetails, setsignUpDetails] = useState({
-    name: "",
+    firstName: "",
+    lastName: "",
     email: "",
     password: "",
   });
   const [errorMessage, setErrorMessage] = useState({
     show: false,
     message: "",
   });
+  const emailRegex =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  const passwordRegex = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*\W).{8,}$/;
+  const { signUpAuth } = useAuth();
+
+  const signUpHandler = () => {
+    if (
+      signUpDetails.firstName !== "" &&
+      signUpDetails.lastName !== "" &&
+      signUpDetails.email.match(emailRegex) &&
```

Comment:
suggestion: destructure the signup details, as you are using it many times below in the code as well